raiden/api/rest.py

```diff
@@ -94,88 +105,135 @@ class RestAPI(object):
 
         # wrap in ChannelList:
-        channel_list = ChannelList(api_result)
+        channel_list = ChannelList(raiden_service_result)
         result = self.channel_list_schema.dumps(channel_list)
         return result
 
+    def get_tokens_list(self):
+        raiden_service_result = self.raiden_api.get_tokens_list()
+        assert isinstance(raiden_service_result, list)
+
+        new_list = []
+        for result in raiden_service_result:
+            new_list.append({'address': result})
+        # wrap in the list Schema
+        tokens_list = TokensList(new_list)
+        result = self.tokens_list_schema.dumps(tokens_list)
+        return result
+
     def get_new_events(self):
-        api_result = self.get_new_events()
-        assert isinstance(api_result, list)
+        raiden_service_result = self.get_new_events()
+        assert isinstance(raiden_service_result, list)
 
         # wrap in EventsList:
-        events_list = EventsList(api_result)
+        events_list = EventsList(raiden_service_result)
         result = self.events_list_schema.dumps(events_list)
         return result
 
-    def patch_channel(self, channel_address, deposit=None, status=None):
-
-        if deposit is not None and status is None:
-            api_result = self.raiden_api.deposit(deposit)
-            return self.channel_schema.dumps(api_result)
-
-        elif status is not None and deposit is None:
-
-            if status == 'closed':
-                api_result = self.raiden_api.close(channel_address)
-                return self.channel_schema.dumps(api_result)
+    def get_channel(self, channel_address):
+        channel = self.raiden_api.get_channel(channel_address)
+        return self.channel_schema.dumps(channel)
+
+    def get_partners_by_token(self, token_address):
+        return_list = []
+        raiden_service_result = self.raiden_api.get_channel_list(token_address)
+        for result in raiden_service_result:
+            return_list.append({
+                'partner_address': result.partner_address,
+                'channel': '/api/{}/channels/{}'.format(
+                    self.version,
+                    address_encoder(result.channel_address)
+                )
+            })
+        # wrap in the list schema
+        schema_list = PartnersPerTokenList(return_list)
+        result = self.partner_per_token_list_schema.dumps(schema_list)
+        return result
 
-            if status == 'settled':
-                api_result = self.raiden_api.close(channel_address)
-                return self.channel_schema.dumps(api_result)
+    def patch_channel(self, channel_address, balance=None, state=None):
+        # find the channel
+        channel = self.raiden_api.get_channel(channel_address)
 
-            if status == 'open':
+        # if we patch with `balance` it's a deposit
+        if balance is not None and state is None:
+            raiden_service_result = self.raiden_api.deposit(
```

Comment:
This can be problematic, we should be really using `balance` here instead of a `deposit`. A client with a timeout for error detection could make more than one request, each new request will unknowingly depositing additional token, this is bad for error detection/recovery.

This should be changed for setting the balance instead of doing an additional deposit.